src/dune/findlib/findlib.ml

```diff
@@ -211,23 +211,34 @@ module Package = struct
     Vars.get_words t.vars "ppx_runtime_deps" preds
     |> List.map ~f:(Lib_name.of_string_exn ~loc:None)
 
-  (* We need a way to figure out if a library is a ppx rewriter. To do that, we
-     currently rely on the presence of [ppx_driver] predicates which Dune uses
-     when the library's kind is not [Normal]. A cleaner way would be to add a
-     new variable to the generated META file, e.g. called [library_kind], but
-     this seems like an overkill given that we plan to stop supporting generated
-     META files soon. *)
+  (* If the META file contains no information about the library kind, we guess
+     the kind by looking at the presence of [ppx_driver] predicates which Dune
+     uses when the library's kind is not [Normal]. *)
   let kind t =
-    match String.Map.find t.vars "requires" with
-    | None -> Lib_kind.Normal
-    | Some ({ set_rules; add_rules } : Rules.t) ->
-      if
-        List.exists (set_rules @ add_rules) ~f:(fun (rule : Rule.t) ->
-            Ps.mem rule.preds_required P.ppx_driver)
-      then
-        Ppx_rewriter Lib_kind.Ppx_args.empty
-      else
-        Lib_kind.Normal
+    match Vars.get t.vars "library_kind" Ps.empty with
+    | Some "normal" -> Lib_kind.Normal
+    | Some "ppx_rewriter" -> Ppx_rewriter Lib_kind.Ppx_args.empty
+    | Some "ppx_deriver" -> Ppx_deriver Lib_kind.Ppx_args.empty
+    | Some s ->
+      User_error.raise [ Pp.textf "Unknown \"library_kind\" setting %S." s ]
```

Comment:
This error should probably mention the name of the library. 
Also, when does this error happen? (if a library of an unknown kind ends up installed, dune should keep working as far as unrelated libraries are concerned)

Reply:
It's important for dune to tolerate META files as much as possible. This is our only window with the non dune world, so we should be quite permissive here. So we should definitely not error out if some field is missing.

Reply:
I removed the error. We now consider the library's kind to be `Normal` if we come across an unknown setting.